utilities/exceptions.py

```
@@ -112,6 +112,16 @@ class UnsupportedCPUArchitectureError(Exception):
     """Exception raised when a CPU architecture is not supported."""
 
 
+class MigrationStuckSchedulingError(Exception):
+    """Exception raised when a migration is stuck in Scheduling state for more than 4 minutes."""
+
+    def __init__(self, migration_name):
```

Ruff (0.15.14)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)